### PR TITLE
Properly handle errors for SimpleFIN.

### PR DESCRIPTION
**Pull request by psybers**
This PR tries to provide better error handling for SimpleFIN.  First, sometimes the calls were never returning a response (they just `return` from the method) so this fixes that issues, which notifies the client side that something happened.

Second, it tries to provide better error messages on the client side.

### CODE REVIEW

**Comment by psybers** [comment]
Related https://github.com/actualbudget/actual/pull/2891

File: src/app-simplefin/app-simplefin.js [hunk 1]
```diff
@@ -2,150 +2,198 @@ import express from 'express';
...
+app.post(
+  '/accounts',
+  handleError(async (req, res) => {
+    let accessKey = secretsService.get(SecretName.simplefin_accessKey);
+
+    try {
+      if (accessKey == null || accessKey === 'Forbidden') {
+        const token = secretsService.get(SecretName.simplefin_token);
+        if (token == null || token === 'Forbidden') {
+          throw new Error('No token');
```
Comment (MatissJanis) [inline src/app-simplefin/app-simplefin.js:35]: **💬 suggestion:** ‏(comment applicable to all instances of `thow` within this file)

Would you mind either adding a prefix to the message OR creating a new Error class? If these errors happen - they will be printed in the error log. At which point it's easier to follow them if a bit more context is added.

```suggestion
          throw new Error('SimpleFIN: no token provided');
```

**Review by MatissJanis — COMMENTED** [review]
Thanks for your work on this! Just one small comment.

File: src/app-simplefin/app-simplefin.js (hunk 1 above)
Comment (psybers) [reply src/app-simplefin/app-simplefin.js:35]: Even if they are catch?  LIne 44 below catches these errors and actually re-throws a different error (with a better log message).

Comment (psybers) [reply src/app-simplefin/app-simplefin.js:35]: (Or more accurately, it catches these and then returns an error response to the client and does not throw anything.)

Comment (MatissJanis) [reply src/app-simplefin/app-simplefin.js:35]: Good comment. But this raises another point - could we have different error messages for each of the two different errors? The more granularity we have in error handling - the less support tickets we will get with confused users.

Comment (psybers) [reply src/app-simplefin/app-simplefin.js:35]: In general, that is a good point.  In this particular case though, if there is a problem with their access key *or* their setup token the solution is still the same: get a new setup token.  So here I think a single error message makes most sense.

**Review by MatissJanis — APPROVED** [review]

**Review by iffy — COMMENTED** [review]
I know this is already merged -- someone just pointed me to it, and I thought I'd add some helpful comments.

File: src/app-simplefin/app-simplefin.js [hunk 2]
```diff
@@ -2,150 +2,198 @@ import express from 'express';
...
-      } else {
-        newTrans.booked = true;
-        dateToUse = trans.posted;
+    try {
+      const results = await getTransactions(accessKey, new Date(startDate));
+
+      const account = results.accounts.find((a) => a.id === accountId);
+
+      const needsAttention = results.errors.find(
+        (e) => e === `Connection to ${account.org.name} may need attention`,
```
[the remark this answers is not in the document]
Comment (MatissJanis) [reply src/app-simplefin/app-simplefin.js:82]: Great! Could you update us when/if you implement this in Simplefin. <3

[the remark this answers is not in the document]
Comment (psybers) [reply src/app-simplefin/app-simplefin.js:82]: > Someone pointed me to this PR that now surfaces errors to users. This is great! Seeing this string matching, though, makes me think that SimpleFIN ought to return published error codes in addition to error messages. I'll try to make that happen soon.

@iffy Thanks!  Honestly what would be the most helpful is if the errors were contextualized to the account data itself.  So then you would know which error matches which specific account(s).

File: src/app-simplefin/app-simplefin.js [hunk 3]
```diff
@@ -2,150 +2,198 @@ import express from 'express';
...
+      );
+      if (needsAttention) {
+        res.send({
+          status: 'ok',
+          data: {
+            error_type: 'ACCOUNT_NEEDS_ATTENTION',
+            error_code: 'ACCOUNT_NEEDS_ATTENTION',
+            status: 'rejected',
+            reason:
+              'The account needs your attention at <a href="https://beta-bridge.simplefin.org/auth/login">SimpleFIN</a>.',
```
[the remark this answers is not in the document]
Comment (psybers) [reply src/app-simplefin/app-simplefin.js:92]: Thanks, I'll fix this in another PR!

File: src/app-simplefin/app-simplefin.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (iffy) [reply src/app-simplefin/app-simplefin.js:82]: That's a good idea. I'll ping you when it happens.

File: src/app-simplefin/app-simplefin.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (psybers) [reply src/app-simplefin/app-simplefin.js:92]: PR: #385

**Pull request merged**